rules/properImportRequireRule.ts

```diff
@@ -0,0 +1,136 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as Lint from 'tslint';
+import * as ts from 'typescript';
+
+// True if es6 module
+const module_cache: { [k: string]: boolean } = {};
+
+function findModulePath(module_name: string): string {
+  let current_dir = process.cwd();
+  if (module_name.startsWith('./')) return ''; // Currently doesn't support local modules.
+
+  while (true) {
+    const current_path = path.join(current_dir, 'node_modules', module_name);
+    if (fs.existsSync(current_path)) return current_path;
+
+    const next_dir = path.dirname(current_dir);
+    if (current_dir === next_dir) return '';
+    current_dir = next_dir;
+  }
+}
+
+// The walker takes care of all the work.
+class ProperImportRequire extends Lint.RuleWalker {
+  private createCustomFailure(es6: boolean, start: number, end: number, variable_name: string, module_name: string, quote: string) {
+    if (es6) {
+      const replacement = `import * as ${variable_name} from ${quote}${module_name}${quote};`;
+      return this.createFailure(
+        start,
+        end,
+        `${module_name} is using ES6 exports. Use ${replacement} instead.`,
+        Lint.Replacement.replaceFromTo(start, end, replacement)
+      )
+    }
+
+    const replacement = `import ${variable_name} = require(${quote}${module_name}${quote});`;
+    return this.createFailure(
+      start,
+      end,
+      `${module_name} is not using ES6 exports. Use ${replacement} instead.`,
+      Lint.Replacement.replaceFromTo(start, end, replacement)
+    )
+  }
+
+  public visitImportDeclaration(node: ts.ImportDeclaration) {
+    const module_name = node.moduleSpecifier.getText().replace(/("|')/g, '');
+    if (module_cache[module_name]) {
+      return super.visitImportDeclaration(node);
+    }
+
+    const variable_name = ((node.importClause as ts.ImportClause).namedBindings as ts.NamespaceImport).name.escapedText as string;
+    const quote = node.moduleSpecifier.getText()[0];
+
+    if (module_cache[module_name] === false) {
+      this.addFailure(this.createCustomFailure(
+        false,
+        node.getStart(),
+        node.getWidth(),
+        variable_name,
+        module_name,
+        quote
+      ));
+      return super.visitImportDeclaration(node);
+    }
+
+    const module_path = findModulePath(module_name);
+    if (module_path) {
+      const pkg = require(path.join(module_path, 'package.json'));
+      const main_file = fs.readFileSync(path.join(module_path, pkg.main), 'utf8');
```

Comment:
No async support?

Reply:
`tslint` methods don't wait for the promises to resolve if using async/await.